### DISCUSSION TITLE
ImageMagickObject in IIS Not Terminating

### DISCUSSION THREAD

**Discussion by jmdl1983** [ImageMagick/ImageMagick#6371] [Help]
### ImageMagick version

7.1.0-6

### Operating system, version and so on

Windows Server 2019

### Description

I understand support for ImageMagickObject.dll has been removed (see #4820), but hopefully someone in the forums here can assist me.

I have a site running on classic ASP which utilizes ImageMagickObject.dll.  This has worked beautifully for years on 2012, but trying to move to 2019: I have an odd error.

1. the script properly initializes the script
`Set Img = Server.CreateObject("ImageMagickObject.MagickImage")`
2. the script properly converts the image from HEIC to JPG
`Img.Convert FOLDER & FILE, FOLDER & Replace(FILE,"heic","jpg")`
3. the script next then fails to delete the original file, because it believes it to still be in use by IIS
`fs.DeleteFile FOLDER & FILE`

In order to be able to remove the file, I have to Restart the Application Pool for this site, and I never had to do this before.

Any ideas why it is not releasing the original file?  I have tried `Set Img = Nothing`, but unfortunately that does not help even.

Thanks

**Comment by jmdl1983** [comment]
As a follow-up to my original post: I had the idea that perhaps it was a problem with my source file as this only seemed to happen with HEIC images.  Sure enough, I tried 15 other random HEIC files from the internet and 14 of the 15 processed fine.  I will attach one of the HEIC files, can anyone see a problem in the image encoding which would break ImageMagick?

[bad-heic.zip](https://github.com/ImageMagick/ImageMagick/files/11587173/bad-heic.zip)

**Comment by dlemstra** [comment]
We have stopped support for `ImageMagickObject.MagickImage`, I would advise you to find an alternative.

**Reply by jmdl1983** [reply]
Thanks I understand this, was looking for a stopgap measure in the interim.  Our website will be remade over the next 1-2 years in ASP.NET but being a corporate site; changes come slow.
